fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/IndexKeyValueToPartialRecord.java

```diff
@@ -93,6 +95,67 @@ void copy(@Nonnull Descriptors.Descriptor recordDescriptor, @Nonnull Message.Bui
                   @Nonnull IndexEntry kv);
     }
 
+    /**
+     * The copier to populate the lucene auto complete suggestion as a value for the field where it is indexed from.
+     * So the suggestion can be returned as a {@link com.apple.foundationdb.record.provider.foundationdb.FDBQueriedRecord} for query.
+     */
+    static class LuceneAutoCompleteCopier implements Copier {
+        private LuceneAutoCompleteCopier() {
+        }
+
+        @Override
+        public void copy(@Nonnull Descriptors.Descriptor recordDescriptor, @Nonnull Message.Builder recordBuilder,
+                         @Nonnull IndexEntry kv) {
+            Tuple keyTuple = kv.getKey();
+            if (keyTuple.size() < 2) {
+                throw new RecordCoreException("Invalid key tuple for auto-complete suggestion's index entry")
+                        .addLogInfo(LogMessageKeys.KEY_SIZE, keyTuple.size())
+                        .addLogInfo(LogMessageKeys.RECORD_TYPE, recordDescriptor.getName());
+            }
+            String fieldName = (String) keyTuple.get(keyTuple.size() - 2);
```

Comment:
Is there a reason you're not pulling the field from the index entry and instead pulling it from the key tuple? The kv has a field definition in it.

Reply:
I am not sure I understand where to get the field definition from the ```IndexEntry```. And this passed ```IndexEntry``` is actually a dummy one to include the grouping keys, field name, suggestion and suggestion value. I had a discussion with @MMcM and we agreed to put the keys, field name and suggestion into the key and the suggestion value into the value, although this is arbitrary in some degree.

Reply:
That makes sense. :) Its pretty arbitrary